README.md

```diff
@@ -0,0 +1,228 @@
+# Cookies and sessions
+
+## Overview
+
+Cookies are a way for an HTTP server to ask the user's browser to store a little bit of data for it, and then get that data back from the browser later.
+
+They are fundamental to the operation of nearly every contemporary website.
+
+Primarily, cookies are used for log in. They provide a way for us to verify who a user is once, and then remember it for their entire session. Without cookies, you would have to provide your username and password on every single request to the server.
+
+Cookies may also be used to store other information about a user, such as what's in their shopping cart, or what ads you've shown them during their visit.
+
+In this document, we'll cover what cookies are, how they fit into the HTTP response flow, and how you can access them within your Rails application.
+
+## Objectives
+  1. Describe a cookie.
+  2. Explain the data flow problem cookies solve.
+  3. Find all the cookies on a page.
+  4. Explain what a session is, as far as Rails is concerned.
+  5. Describe where session data is stored.
+  6. Write data to the Rails session and access it later
+     in the app.
+
+## Shopping
+
+Let's say we want to build an e-commerce site. Users can come to the site, add some items to their shopping cart, and then pay for all of them at the end. Simple.
+
+Where does the user's shopping cart live?
+
+We could create a `Cart` model, which `has_and_belongs_to_many` `Item`s. But that leaves a fundamental problem: how do we know what cart to load when we get a request?
```

Comment:
let's link to the documentation on `has_and_belongs_to_many` here. i'm not sure if students will have seen this association yet. http://guides.rubyonrails.org/association_basics.html#the-has-and-belongs-to-many-association